### PR TITLE
Jetpack plugins: tweak browser design

### LINKED ISSUE #11393: Plugins: Space to wrap plugin name
screen shot 2017-02-14 at 1 22 05 pm

It seems like there is enough space to wrap plugin names to two lines before truncating them, without causing the container size to grow. This would improve usability, often the plugin name is cut off and you are missing an important piece of information.

### PR DESCRIPTION
**Pull request by drw158**
fixes #11393 

Biggest change is that now the plugin titles do not truncate.

screen shot 2017-02-16 at 12 18 11 pm

Changes:

- flex layout so that grids items can expand vertically
- bigger text and plugin icons
- increase text contrast
- remove text truncation
- `See All` button now uses standard 18px gridicon

cc @MichaelArestad 

Before:

8bivpqqqxafngaaaaasuvork5cyii
wgsma4vqpvivqaaaabjru5erkjggg

After:

wnbvh fwqocljletlxaaaaaelftksuqmcc
wkyx6kasm5 sqaaaabjru5erkjggg

[Here's the design before the wide layout](https://cloud.githubusercontent.com/assets/618551/23034453/7c90cd34-f441-11e6-8915-1c4afa73a791.png)

### CODE REVIEW

**Comment by Copons** [comment]
There's a problem with the `<a>` size of "non wrapping" plugins when there's a "wrapping" one in the same row.




Basically, the link is sort of "absolutely" sized, and its height depends on its own title/author height.
If one plugin in a row is bigger, all boxes are sized accordingly but the links aren't.
If you try to click below the link "overlay", nothing happens (but the cursor is still a pointer, causing possible confusion).

Maybe it could make sense to change the `.plugins-browser-item__link` link into a full fledged `flex` container?

**Comment by MichaelArestad** [comment]
Why remove the text truncation? I think it looks pretty rough when wrapping and isn't super necessary.

I also think there's a bit too much space between the stars/installed label and the icon/text above it. I would optically align this even if it technically breaks our grid rules.

File: client/my-sites/plugins/plugins-browser-item/style.scss [hunk 1]
```diff
@@ -15,7 +14,7 @@
 	}
 
 	.plugins-browser-item__link {
-		padding: 16px;
+		padding: 16px 16px 48px 16px;
```
Comment (MichaelArestad) [inline client/my-sites/plugins/plugins-browser-item/style.scss]: Could be written as `padding: 16px 16px 48px;` (top, left/right, bottom)

Comment (drw158) [inline client/my-sites/plugins/plugins-browser-item/style.scss]: Whoops, thanks.

**Comment by drw158** [comment]
@Copons Darn, I thought I could get away with not making it a flex container. Seems like that is the best way to go though.

@MichaelArestad I think we should eliminate text truncation if possible, because it's not a great solution. It might look more tidy, but that shouldn't come at the expense of cutting off the name. I do think there is a reason to have an upper limit (some are quite long), but at the moment a large percentage of names are being cut off. #11393 

There might be a way to not truncate the text _and_make it look good. The wp-admin plugin browser looks fine.

We could limit it to 2 lines, but the solution I found was complex. Might still be worth doing. https://github.com/Automattic/wp-calypso/issues/11393

**Comment by MichaelArestad** [comment]
@drw158 I think one line is enough, especially with your removal of the max width on the browser. Users can get a pretty good idea before the cutoff what the plugin is about and can click it for the full name.

**Comment by drw158** [comment]
@apeatling Do you think the truncation is an issue now that the browser uses a wide layout?

**Comment by drw158** [comment]
- Brought back text truncation
- decreased space above rating
- because the name does not wrap anymore, the size of the `<a>` is not an issue

screen shot 2017-02-20 at 3 56 30 pm

**Comment by MichaelArestad** [comment]
💯

**Comment by Copons** [comment]
👍

**Comment by apeatling** [comment]
I think at some breakpoints the truncating makes the plugin browser usability really poor. At bigger screen sizes it's probably okay, although I still think the truncating is prioritizing what looks good, versus what works the best for users.

Example of a breakpoint where this looks bad:

screen shot 2017-02-21 at 11 03 49 am

**Comment by MichaelArestad** [comment]
@apeatling I can't remember if we're using flexbox or how I built that thing, but that's easily resolvable by changing their `flex-basis` to have a min width and encourage wrapping so it doesn't get janky like that on any breakpoint.

**Comment by drw158** [comment]
Easiest solution is to trigger the single column earlier, since it's not using flex box.

nln5exx3mnxg2wnoaaaaaelftksuqmcc

This is the most truncation, after the breakpoint adjustment:

szdky50maaqaaaabjru5erkjggg

If this is ok, then I'll push that breakpoint adjustment. We could tweak the actual breakpoint, but afaik we aren't suppose to create new breakpoints. Another alternative is to use flex box, and the boxes would resize smartly without breakpoints (I think).

**Comment by dmsnell** [comment]
I don't think it's a problem but I did have a little bit of truncating on long titles on mobile (instead of wrapping)

fullsizeoutput_1160

**Comment by roundhill** [comment]
Code LGTM 🚢

**Pull request merged**